vm/actor/src/builtin/market/mod.rs

```diff
@@ -476,6 +469,176 @@ impl Actor {
 
         Ok(commd)
     }
+
+    fn cron_tick<BS, RT>(rt: &mut RT) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*CRON_ACTOR_ADDR))?;
+        let mut amount_slashed = BigUint::zero();
+        let mut timed_out_verified_deals: Vec<DealProposal> = Vec::new();
+
+        rt.transaction::<State, Result<(), ActorError>, _>(|st, rt| {
+            let mut dbe =
+                SetMultimap::from_root(rt.store(), &st.deal_ops_by_epoch).map_err(|e| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("failed to load deal opts set: {}", e),
+                    )
+                })?;
+
+            let mut updates_needed: Vec<(ChainEpoch, DealID)> = Vec::new();
+
+            let mut states = Amt::load(&st.states, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let mut et = BalanceTable::from_root(rt.store(), &st.escrow_table)
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let mut lt = BalanceTable::from_root(rt.store(), &st.locked_table)
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let mut i = st.last_cron + 1;
+            while i <= rt.curr_epoch() {
+                dbe.for_each(i, |id| {
+                    let mut state: DealState = states
+                        .get(id)
+                        .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))
+                        .unwrap()
+                        .ok_or(format!("could not find deal state: {}", id))
+                        .unwrap();
+
+                    let deal = st.must_get_deal(rt.store(), id).unwrap();
+                    // Not yet appeared in proven sector; check for timeout.
+                    if state.sector_start_epoch.is_none() {
+                        assert!(
+                            rt.curr_epoch() >= deal.start_epoch,
+                            "if sector start is not set, we must be in a timed out state"
+                        );
+
+                        let slashed = st
+                            .process_deal_init_timed_out(
+                                rt.store(),
+                                &mut et,
+                                &mut lt,
+                                id,
+                                &deal,
+                                state,
+                            )
+                            .unwrap();
```

Comment:
again, don't unwrap, they return with a valid exit code if this fails (fn should return an ActorError and this should just try return